### ISSUE TITLE
Allow multiple controllers

### ISSUE THREAD

**Issue by michaelsproul** [sigp/eleel#29]
It could be useful to allow multiple CLs to connect to the `/canonical` endpoint.

Currently this is _possible_, but it won't necessarily work particularly well if the CLs send conflicting updates. It's basically the same as if the controlling CLs were directly connected the EL.

Instead, Eleel could act as a mediator, detecting conflicts and preventing them from reaching the EL. The hard part would be tracking _which_ CL to follow in the event of a conflict. This requires some state beyond just "which CL sent me an update first", which would have the EL re-orging all over the place.

**Comment by Savid** [comment]
would the primary use case for this being able to restart one of the "canonical" CL gracefully and not impacting other non canonical CLs attached? If so this would be an awesome feature!

**Comment by michaelsproul** [comment]
@Savid Yeah, exactly. It would provide redundancy + resilience for all the CLs sharing the EL.

I'm not sure if I'll have time to work on this _soon_, but could help mentor a PR if you wanted to give it a shot

**Comment by Savid** [comment]
> I'm not sure if I'll have time to work on this _soon_, but could help mentor a PR if you wanted to give it a shot

Yep sounds good, can have a crack in the next week or two
